### DISCUSSION TITLE
How to change query parameters while keeping the scroll position

### DISCUSSION THREAD

**Discussion by dninomiya** [facebook/docusaurus#7114] [Q&A] [answered]
When using useHistory, changing only the query parameters for the same path will return the scroll position to the top.

**Comment by Josh-Cena** [comment, marked as answer]
When pushing or replacing history, you need to save the scroll position as state. See our own implementation on the showcase page: 

https://github.com/facebook/docusaurus/blob/e610a4ac002e561b8adee5558b178168a2e26742/website/src/pages/showcase/index.tsx#L48-L67

**Reply by dninomiya** [reply]
solved. thank you!
